titiler/pgstac/factory.py

```diff
@@ -127,33 +82,15 @@ class MosaicTilerFactory(BaseFactory):
     backend: Type[PGSTACBackend] = PGSTACBackend
     backend_dependency: Type[DefaultDependency] = BackendParams
 
+    # Backend.get_assets() Options
+    assets_accessor_dependency: Type[DefaultDependency] = PgSTACParams
```

Comment:
this is a new Factory attribute in `titiler.mosaic.factory.MosaicTilerFactory` which pass options to the `.tile` methods (and gets forwarded to the `get_assets()` method. 

so because this is now part of the endpoint method we don't need custom code here to pass the `PgSTACParams` params